src/server/gc_worker/applied_lock_collector.rs

```diff
@@ -382,10 +384,14 @@ impl Runnable for LockCollectorRunner {
 pub struct AppliedLockCollector {
     worker: Mutex<Worker>,
     scheduler: Scheduler<LockCollectorTask>,
+    concurrency_manager: ConcurrencyManager,
```

Comment:
Feels we've made many things coupled and there may still be some possible corner cases.